numba/tests/test_typedlist.py

```diff
@@ -155,6 +155,19 @@ def test_unsigned_access(self):
         self.assertEqual(L.pop(ui32_1), 2)
         self.assertEqual(L.pop(ui32_0), 123)
 
+    def test_dtype(self):
```

Comment:
You should try to figure out how `dtype` might become available within `@njit` functions:

```python
@njit 
def foo():
    l = List()
    l.append(1)
    return l.dtype
```

Or something along those lines. And then add some tests for that if possible.

Reply:
Got it, I will add tests for this as well

Reply:
I was thinking about this again last night, I think you may need an overload for this.

Reply:
Okay thanks! I will look into that.

Reply:
`dtype` can now be used within `@njit` functions. Please check it out! :D